conjure-lib/src/main/java/com/palantir/conjure/java/lib/internal/ConjureCollections.java

```diff
@@ -158,11 +165,14 @@ public static List<Double> newNonNullDoubleList(Iterable<Double> iterable) {
         return doubleList;
     }
 
-    /**
-     * The following Conjure boxed list wrappers for the eclipse-collections [type]ArrayList are temporary (except
-     * ConjureSafeLongList). In eclipse-collections 12, a BoxedMutable[type]List will be released. Once available,
-     * Conjure[type]List should be replaced with that.
-     */
+    // This method modifies a list that can't handle nulls. Do not use this unless the nonNullCollections flag is set
+    public static void addAllToDoubleList(Collection<Double> addTo, double[] elementsToAdd) {
+        if (addTo instanceof ConjureDoubleList) {
```

Comment:
An inelegant solution to preserve the feedback on https://github.com/palantir/conjure-java/pull/2307#discussion_r1598822523, but still capture the optimization we added. Also has [some prior art in `addAll`.](https://github.com/palantir/conjure-java/blob/ae51640c5a77e5dc663750fb0e943aefc2a0dcd4/conjure-lib/src/main/java/com/palantir/conjure/java/lib/internal/ConjureCollections.java#L45)